src/main/java/main/internet/shop/dao/jdbcimpl/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,171 @@
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (!resultSet.next()) {
+                return Optional.empty();
+            }
+            Order order = extractOrderFromResultSet(resultSet);
+            statement.close();
+            order.setProducts(extractProductsFromOrder(order.getId()));
+            return Optional.of(order);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't' get order from DB with Id = " + id, e);
+        }
+    }
+
+    @Override
+    public List<Order> getAll() {
+        List<Order> orders = new ArrayList<>();
+        String query = "SELECT * FROM orders WHERE deleted = false";
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                orders.add(extractOrderFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get orders from DB", e);
+        }
+        for (Order order : orders) {
+            order.setProducts(extractProductsFromOrder(order.getId()));
+        }
+        return orders;
+    }
+
+    @Override
+    public Order update(Order order) {
+        String query = "DELETE FROM orders_products WHERE order_id = ?;";
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, order.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete product from order with id = "
+                    + order.getId(), e);
+        }
+        return insertProducts(order);
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "UPDATE orders SET deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete order with ID = " + id, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Order order) {
+        return deleteById(order.getId());
+    }
+
+    private Order extractOrderFromResultSet(ResultSet resultSet) throws SQLException {
+        Order order = new Order(resultSet.getLong("user_id"));
+        order.setId(resultSet.getLong("id"));
+        return order;
+    }
+
+    private List<Product> extractProductsFromOrder(Long id) {
```

Comment:
```suggestion
    private List<Product> getOrderProducts(Long id) {
```
You are making a new request to Db